xblock/fields.py

```diff
@@ -839,18 +867,18 @@ def _sanitize(self, value):
         https://www.w3.org/TR/xml/#charsets
         Leave all other characters.
         """
-        if isinstance(value, unicode):
-            new_value = u''.join(ch for ch in value if unicodedata.category(ch)[0] != u'C' or ch in (u'\n', u'\r', u'\t'))
-        elif isinstance(value, str):
-            new_value = ''.join(ch for ch in value if ord(ch) >= 32 or ch in ('\n', '\r', '\t'))
+        if isinstance(value, six.text_type):
+            new_value = u''.join(ch for ch in value if unicodedata.category(ch)[0] != u'C' or ch in u'\n\r\t')
+        elif isinstance(value, six.binary_type):
+            new_value = ''.join(ch for ch in value if ord(ch) >= 32 or ch in b'\n\r\t')
```

Comment:
I suspect it would be better to handle this by converting the binary value to text first, and only stripping bad characters once.